pkg/git/repository.go

```diff
@@ -158,9 +158,23 @@ func (r *Repository) StageFiles(filenames ...string) error {
 }
 
 // Commit does the git commit -m with the msg & author
+// after first running git config commands for the user.name and user.email
```

Comment:
Hmm, I see. My rationale was that if you're going to become somebody else (say, locally), to set the name and email, you'd use the git config settings. But you can do things on a commit basis (like `git commit --author "foo")

I'm wondering if we can 
1) not do it on a global basis2
2) only do this under Tekton somehow 🤔

Reply:
I'm going to check

```
Does this mean that after running `services promote`, that if I set `--commit-email` or `--commit-name`, that my `~/.gitconfig` will have been amended to use those values?
```

now

Reply:
I don't think we should be changing these files.

I can't imagine you'd want to edit your files to run it locally.

Reply:
I think this is related to my above comment - nobody's suggesting modifying your own $HOME/.gitconfig to run things locally - I'm looking into just omitting that --global flag now.

Reply:
```
[promote : promote] + services promote --commit-name=a-roberts --commit-email=aroberts@uk.ibm.com --from /workspace/repo --to https://github.com/a-roberts/gitops-example-dev.git --service promote-demo --commit-message 'Publish promote-demo commit cc392fc'
[promote : promote] Setting Git config for user.Name and user.Email for /tekton/home/.promotion/cache/https%3A%2F%2Fpromotion%3<mytoken>github.com%2Fa-roberts%2Fgitops-example-dev.git-workspace-repo-local-dir-99315/gitops-example-dev
[promote : promote] 2020/05/22 12:47:45 created PR 8
```

@bigkevmcd @mnuttall so just to be sure, since modifying a user's global $HOME/.gitconfig file or requiring them to fiddle with it would be outrageous ^^

So it's only being done at the repository scope now, from our cache directory

Reply:
Great, thank you, this is much better